src/open_inwoner/openklant/wrap.py

```diff
@@ -87,6 +90,59 @@ def fetch_klantcontactmoment(
     return kcm
 
 
+def _fetch_objectcontactmomenten_for_contactmoment(
+    contactmoment: ContactMoment, *, client=None
+) -> List[ContactMoment]:
+    client = client or build_client("contactmomenten")
+    if client is None:
+        return []
+
+    try:
+        response = get_paginated_results(
+            client,
+            "objectcontactmoment",
+            request_kwargs={"params": {"contactmoment": contactmoment.url}},
+        )
+    except (RequestException, ClientError) as e:
+        logger.exception("exception while making request", exc_info=e)
+        return []
+
+    object_contact_momenten = factory(ObjectContactMoment, response)
+
+    # resolve linked resources
+    for ocm in object_contact_momenten:
+        assert ocm.contactmoment == contactmoment.url
+        ocm.contactmoment = contactmoment
+        if ocm.object_type == "zaak":
+            object_path = urlparse(ocm.object).path
+            ocm.object = fetch_single_case(object_path.split("/")[-1])
```

Comment:
How likely is it we have multiple contact-moments with the same object? Maybe it would be better to de-duplicate the requests as the API's are slow.

Also it is not good pattern to parse the URLs for UUID's; they are supposed to be opaque. Just make a fetch_single_case() variant that uses a URL.

Reply:
> How likely is it we have multiple contact-moments with the same object? Maybe it would be better to de-duplicate the requests as the API's are slow.

I think that it will occur sometimes, I'm not entirely sure what you mean by deduplicating though. This function retrieves the objectcontactmomenten for one specific contactmoment (it's called from the Contactmoment detail view)

Reply:
But in this loop, multiple `ocm`'s could link to the same `ocm.object` right? 

If you create a dict with the URI's as key (and None as value) it naturally de-duplicates the URI's, so then you can load all the unique URI's with `.keys()` and assign each result back to the dict (replacing the None). 

Then go through the `ocm`'s again and grab the resource by URI from the dict.

Reply:
Ahh like that, I'm not sure if it is possible for a single contactmoment to have multiple relations through objectcontactmoment with the same `object` (zaak), but I could add it just to be sure